src/find-options/FindOptionsWhere.ts

```diff
@@ -40,11 +41,20 @@ export type FindOptionsWhereProperty<
           | Property
     : Property | FindOperator<Property>
 
+/**
+ * Helper type to conditionally allow null/undefined based on TypeORMSettings
+ */
+type NullableWhereValue<T> = TypeORMSettings extends {
+    allowNullableWhere: true
+}
+    ? T | null
+    : T
+
 /**
  * Used for find operations.
  */
 export type FindOptionsWhere<Entity> = {
     [P in keyof Entity]?: P extends "toString"
         ? unknown
-        : FindOptionsWhereProperty<NonNullable<Entity[P]>>
+        : NullableWhereValue<FindOptionsWhereProperty<NonNullable<Entity[P]>>>
```

Comment:
I just realised that TypeScript currently _doesn't_ care if you pass an `undefined` value, it only complains about `null` (which contradicts my doc edits!).

This makes using `undefined: "throw"` mode more dangerous than expected.

What do you think about:

```suggestion
        : NullableWhereValue<FindOptionsWhereProperty<Required<NonNullable<Entity[P]>>>>
```

This could be a breaking change. 😞  Maybe we need another setting for it.